Cargo.toml

```diff
@@ -20,3 +20,20 @@ path = "lapce-proxy/src/bin/lapce-proxy.rs"
 
 [workspace]
 members = ["lapce-ui", "lapce-proxy", "lapce-rpc", "lapce-data", "lapce-core"]
+
+# A profile which compiles all dependencies in release mode but lapce
+# code in dev mode. This gives a good debugging experience for your
+# code and fast performance of other people's code. After the initial
+# build subsequent ones are as fast as dev mode builds.
+# See https://doc.rust-lang.org/cargo/reference/profiles.html
+# To use this profile:
+#   cargo build --profile fastdev
+#   cargo run --profile fastdev --bin lapce
+[profile.fastdev.package."*"]
+opt-level = 3
+
+[profile.fastdev.build-override]
```

Comment:
According to the docs, this is "build dependencies (build scripts, proc macros, and their dependencies)". What do we expect from this, compile-time improvements?

Reply:
Also, should we set `build-override` for the release profile as well? According to the docs, build dependencies are never optimized.

Reply:
I'm not sure to be honest, it was new to me. I put it in because Lapce-UI and Data have `[build-dependencies]`, but on closer inspection I'm not sure why - I don't see any build scripts.

Reply:
I don't expect a lot here. The longest build scripts are I believe the 30s per piece compile times for tree-sitter packages, but I don't expect those to improve because of this. I think we should try and compare clean build times with and without this section.

Reply:
I'll have a look later this evening. It seems that the `[build_dependencies]` sections in UI and data can simply be removed, the project builds without them.

Reply:
I think we might want to emphasize that `[profile.fastdev.package."*"]` is only active for **non-workspace** packages, i.e. lapce-data and co. aren't affected. The docs are clear on this, and you already have "lapce code" in there, but it was a lot less obvious for me than it could have.

Reply:
> I think we might want to emphasize that `[profile.fastdev.package."*"]` is only active for **non-workspace** packages, i.e. lapce-data and co. aren't affected. The docs are clear on this, and you already have "lapce code" in there, but it was a lot less obvious for me than it could have.

Comment in Cargo.toml updated.

Reply:
> I'll have a look later this evening. It seems that the `[build_dependencies]` sections in UI and data can simply be removed, the project builds without them.

Removed these sections. They appear to make no difference to compile times, and hence I assume no difference to the generated code.